include/systems/generic_projector.h

```diff
@@ -2573,17 +2585,20 @@ GenericProjector<FFunctor, GFunctor, FValue,
 
 template <typename FFunctor, typename GFunctor,
           typename FValue, typename ProjectionAction>
+template <typename OutputShape>
 void
 GenericProjector<FFunctor, GFunctor, FValue, ProjectionAction>::SubProjector::construct_projection
   (const std::vector<dof_id_type> & dof_indices_var,
    const std::vector<unsigned int> & involved_dofs,
    unsigned int var_component,
    const Node * node,
-   const FEBase & fe)
+   const FEGenericBase<OutputShape> & fe)
 {
-  const std::vector<Real> & JxW = fe.get_JxW();
-  const std::vector<std::vector<Real>> & phi = fe.get_phi();
-  const std::vector<std::vector<RealGradient>> * dphi = nullptr;
+  typedef FEGenericBase<OutputShape> FEBase;
```

Comment:
I know reusing "FEBase" makes the diff a lot shorter, but could we pick a new name to use here?  I'm surprised this doesn't trigger -Wshadow already, and I don't want to be caught out as soon as a compiler gets more thorough about that warning.

Reply:
👍